src/main/java/com/dabsquared/gitlabjenkins/connection/GitlabCredentialResolver.java

```diff
@@ -0,0 +1,36 @@
+package com.dabsquared.gitlabjenkins.connection;
+
+import hudson.model.Item;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+public class GitlabCredentialResolver {
+
+    private Item item;
+    private String credentialsId;
+
+    public GitlabCredentialResolver() {}
+
+    public GitlabCredentialResolver(Item item, String credentialsId) {
+        this.item = item;
+        this.credentialsId = credentialsId;
+    }
+
+    public Item getItem() {
```

Comment:
```suggestion
    @Restricted(NoExternalUse.class)
    public Item getItem() {
```

Reply:
This is just to be safe, since its usage is wrapped around the already restricted `getApiToken` method in "GitLabClientBuilder.java".